src/components/common/Stepper.vue

```diff
@@ -33,11 +33,40 @@ import { Component, Vue } from 'vue-property-decorator'
 import { Getter } from 'vuex-class'
 
 // Interfaces
-import { GetterIF } from '@/interfaces'
+import { RouteNames } from '@/enums'
 
 @Component({})
 export default class Stepper extends Vue {
-  @Getter getSteps!: GetterIF
+  @Getter getSteps!: Array<any>
+  @Getter isBusySaving!: boolean
+  @Getter isDefineCompanyValid!: boolean
+  @Getter isAddPeopleAndRolesValid!: boolean
+  @Getter isCreateShareStructureValid!: boolean
+  @Getter isIncorporationAgreementValid!: boolean
+  @Getter isApplicationValid!: boolean
+
+  /** Returns true if the step route is valid. */
+  private isValid (route: RouteNames): boolean {
+    switch (route) {
+      case RouteNames.DEFINE_COMPANY:
+        return this.isDefineCompanyValid
+
+      case RouteNames.ADD_PEOPLE_AND_ROLES:
+        return this.isAddPeopleAndRolesValid
+
+      case RouteNames.CREATE_SHARE_STRUCTURE:
+        return this.isCreateShareStructureValid
+
+      case RouteNames.INCORPORATION_AGREEMENT:
+        return this.isIncorporationAgreementValid
+
+      case RouteNames.REVIEW_CONFIRM:
+        return this.isApplicationValid
+
+      default:
+        return false
+    }
```

Comment:
So this is an abstraction to handle any possible future steps/pages? Sounds good.

Reply:
This is just for the purposes of validation on each step but yessir you betcha. 
All the rest is contained within the configuration but i think validation will need to be something computed ( obviously. )